src/isArray.ts

```diff
@@ -1,3 +1,5 @@
+type IfIsAny<T, Then, Else> = 0 extends 1 & T ? Then : Else;
```

Comment:
Looks like this is inspired by type-fest; let's put this in types.ts and add some documentation linking to type-fest so we have a point of reference if we ever need one.

Reply:
I wasn't aware type-fest had a is-any utility type. Nice.
Will reference the link to their code 👍